app/decorators/session_decorator.rb

```diff
@@ -38,6 +38,6 @@ def sp_return_url; end
   def requested_attributes; end
 
   def cancel_link_path
-    root_path
+    root_path(locale: I18n.locale)
```

Comment:
We have the helper method `locale_url_param` included in a bunch of places (if not, `include LocaleHelper` here) It has the advantage of passing `nil` when we're in `en` to keep the URLs slightly cleaner?